egs/tedlium/s5/local/chain/run_tdnn.sh

```diff
@@ -0,0 +1,211 @@
+  # once, the second one has zero or more repeats.]
+  rm -rf $lang
+  cp -r data/lang $lang
+  silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [ $stage -le 11 ]; then
+  # Build a tree using our new topology.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --leftmost-questions-truncate $leftmost_questions_truncate \
+      --cmd "$train_cmd" 4200 data/$train_set $lang $ali_dir $treedir
+fi
+
+if [ $stage -le 12 ]; then
+  echo "$0: creating neural net configs";
+  dim_opts="--relu-dim $relu_dim"
+
+  # create the config files for nnet initialization
+  repair_opts=${self_repair_scale:+" --self-repair-scale $self_repair_scale "}
+
+  steps/nnet3/tdnn/make_configs.py \
+    $repair_opts \
+    --feat-dir data/${train_set}_hires \
+    --ivector-dir exp/nnet3/ivectors_${train_set} \
+    --tree-dir $treedir \
+    $dim_opts \
+    --splice-indexes "-1,0,1 -1,0,1,2 -3,0,3 -3,0,3 -3,0,3 -6,-3,0 0" \
+    --use-presoftmax-prior-scale false \
+    --xent-regularize $xent_regularize \
+    --xent-separate-forward-affine true \
+    --include-log-softmax false \
+    --final-layer-normalize-target $final_layer_normalize_target \
+    $dir/configs || exit 1;
+fi
+
+if [ $stage -le 13 ]; then
+  if [[  $(hostname -f) ==  *.clsp.jhu.edu ]]; then
+     # spread the egs over various machines.  will help reduce overload of any
+     # one machine.
+     utils/create_split_dir.pl /export/b{09,10,11,12}/$USER/kaldi-data/egs/tedlium-$(date +'%m_%d_%H_%M')/s5/$dir/egs/storage $dir/egs/storage
+  fi
+
+  touch $dir/egs/.nodelete
+
+ steps/nnet3/chain/train.py --stage $train_stage \
+   --cmd "$decode_cmd" \
+   --feat.online-ivector-dir exp/nnet3/ivectors_${train_set} \
+   --feat.cmvn-opts "--norm-means=false --norm-vars=false" \
+   --chain.xent-regularize $xent_regularize \
+   --chain.leaky-hmm-coefficient 0.1 \
+   --chain.l2-regularize 0.00005 \
+   --chain.apply-deriv-weights false \
+   --chain.lm-opts="--num-extra-lm-states=2000" \
+   --egs.stage $get_egs_stage \
+   --egs.opts "--frames-overlap-per-eg 0" \
+   --egs.chunk-width $frames_per_eg \
+   --trainer.num-chunk-per-minibatch $minibatch_size \
+   --trainer.frames-per-iter 1500000 \
+   --trainer.num-epochs $num_epochs \
+   --trainer.optimization.num-jobs-initial $num_jobs_initial \
+   --trainer.optimization.num-jobs-final $num_jobs_final \
+   --trainer.optimization.initial-effective-lrate $initial_effective_lrate \
+   --trainer.optimization.final-effective-lrate $final_effective_lrate \
+   --trainer.max-param-change $max_param_change \
+   --cleanup.remove-egs $remove_egs \
+   --feat-dir data/${train_set}_hires \
+   --tree-dir $treedir \
+   --lat-dir exp/tri3_lats$suffix \
```

Comment:
once again .. if you are using a variable $gmm then use it in place of every tri3 occurrence in this script.